webapi/src/main/java/org/sakaiproject/webapi/controllers/GradesController.java

```diff
@@ -112,25 +110,27 @@ public class GradesController extends AbstractSakaiApiController {
                     reference = SamigoReferenceReckoner.reckoner().site(siteId).subtype("p").id(a.getExternalId()).reckon().getReference();
                 }
 
-                int totalGradableUserCount = a.getExternallyMaintained() ?
-                        entityManager.getEntity(reference).map(externalEntity -> {
-
-                            return externalEntity.getGroupReferences().map(groupReferences -> {
-
-                                if (groupReferences.isEmpty()) return siteUserIds.size();
+                int totalGradableUserCount = siteUserIds.size();
 
-                                return groupReferences.stream().reduce(0, (t, gr) -> {
-
-                                    try {
-                                        return t + getGradableUsers(authzGroupService.getAuthzGroup(gr)).size();
-                                    } catch (Exception e) {
-                                        log.warn("No authz group found for reference {}", gr);
-                                    }
-                                    return t;
-                                }, Integer::sum);
-                            }).orElse(siteUserIds.size());
-                        }).orElse(0)
-                    : siteUserIds.size();
+                if (a.getExternallyMaintained()) {
+                    Optional<Entity> optionalEntity = entityManager.getEntity(reference);
+                    if (optionalEntity.isEmpty() || optionalEntity.get().getGroupReferences().isEmpty()) continue; // entity is not accessible, so skip
+
+                    Entity entity = optionalEntity.get();
+                    totalGradableUserCount = entity.getGroupReferences()
+                            .filter(refs -> !refs.isEmpty())
+                            .map(refs -> refs.stream()
+                                    .mapToInt(g -> {
+                                        try {
+                                            return getGradableUsers(authzGroupService.getAuthzGroup(g)).size();
+                                        } catch (Exception e) {
+                                            log.warn("No authz group found for reference {}", g);
+                                            return 0;
+                                        }
+                                    })
+                                    .sum())
+                            .orElse(siteUserIds.size());
```

Comment:
this orElse seems unnecessary as it is already being set above on line 113?

Reply:
yeah it appears that away and even AI was confused, but I debugged this lambda thoroughly. The key is that were dealing with Optional(s) and a Set, the Optional can be empty as well as a Set can be empty that's why you see those.